agent/structs/structs.go

```diff
@@ -1554,6 +1554,20 @@ func (nodes CheckServiceNodes) Shuffle() {
 	}
 }
 
+func (nodes CheckServiceNodes) ToServiceDump() ServiceDump {
+	var ret ServiceDump
+	for i := 0; i < len(nodes); i++ {
```

Comment:
Why not just use `range` here?

Reply:
Forgot about this syntax:
```suggestion
	for i := range nodes {
```

Will add it here.